### PR TITLE
Add miniaod to validation

### PR DESCRIPTION
**Pull request by dntaylor**
This is a redo on top of slava77/master with comments from #77 taken into account.

For the matrix_70X.txt: I'm not certain of all the file names (and which ones run miniAOD and which don't). So I only added the miniAOD part to the tests that ran successfully in CMSSW_8_0_0_pre5.

For the packed candidates I get an error when running the packedCand function.

```
plotting variable: patPackedCandidates_packedPFCandidates__RECO.obj.charge()
Error in <TTreeFormula::DefinedVariable>: Can not call method charge() on class without dictionary (pat::PackedCandidate)!
Error in <TTreeFormula::Compile>:  Part of the Variable "patPackedCandidates_packedPFCandidates__RECO.obj.charge()" exists but some of it is not accessible or useable
Info in <TSelectorDraw::AbortProcess>: Variable compilation failed: {patPackedCandidates_packedPFCandidates__RECO.obj.charge(),}
Comparison died
```

### CODE REVIEW

**Comment by cvuosalo** [comment]
@dntaylor: In general, this PR looks good. Concerning the error, do the other packed candidate quantities work, or does the failure of the `charge` comparison terminate all packed candidate comparisons? Might the problem be a typo or spelling error of the variable name?

File: validate.C [hunk 1]
```diff
@@ -1194,67 +1379,7 @@ void validateEvents(TString step, TString file, TString refFile, TString r="RECO
...
-      recoMuons("pfMeanDRIsoProfileR03().sumChargedHadronPt", true);
-      recoMuons("pfMeanDRIsoProfileR03().sumChargedParticlePt", true);
-      recoMuons("pfMeanDRIsoProfileR03().sumNeutralHadronEt", true);
-      recoMuons("pfMeanDRIsoProfileR03().sumPhotonEt", true);
-      recoMuons("pfMeanDRIsoProfileR03().sumPUPt", true);
-      recoMuons("pfSumDRIsoProfileR03().sumChargedHadronPt", true);
-      recoMuons("pfSumDRIsoProfileR03().sumChargedParticlePt", true);
-      recoMuons("pfSumDRIsoProfileR03().sumNeutralHadronEt", true);
-      recoMuons("pfSumDRIsoProfileR03().sumPhotonEt", true);
-      recoMuons("pfSumDRIsoProfileR03().sumPUPt", true);
```
Comment (slava77) [inline validate.C]: old lines 1247 to 1253 do not have a corresponding code in the new version

File: validate.C [hunk 2]
```diff
@@ -291,63 +289,98 @@ void calomet(TString algo, TString var, bool doLog10 = false){
...
-void photon(TString var, TString cName = "photons_", bool notafunction=false){
-  TString v= notafunction ? "recoPhotons_"+cName+"_"+recoS+".obj."+var :
-    "recoPhotons_"+cName+"_"+recoS+".obj."+var+"()" ;
+void tauVars(TString cName = "hpsPFTauProducer_", TString tName = "recoPFTaus_"){
+  plotvar(tName+cName+"_"+recoS+".obj@.size()");
+  tau("energy",cName,tName);
+  tau("et",cName,tName);
+  tau("eta",cName,tName);
+  tau("phi",cName,tName);
+  if (tName!="patTaus_") tau("emFraction",cName,tName);
```
Comment (slava77) [inline validate.C]: is it going to crash if emFraction is tried on patTaus?
It's easier to maintain without the extra ifs

File: validate.C [hunk 3]
```diff
@@ -291,63 +289,98 @@ void calomet(TString algo, TString var, bool doLog10 = false){
...
+  photon("showerShapeVariables().e1x3",cName,tName, true);
+  photon("showerShapeVariables().e2x2",cName,tName, true);
+  photon("showerShapeVariables().e2x5Max",cName,tName, true);
+  photon("showerShapeVariables().e2x5Left",cName,tName, true);
+  photon("showerShapeVariables().e2x5Right",cName,tName, true);
+  photon("showerShapeVariables().e2x5Top",cName,tName, true);
+  photon("showerShapeVariables().e2x5Bottom",cName,tName, true);
 
   //  photon("r9", cName);
   photon("mipChi2", cName);
```
Comment (slava77) [inline validate.C]: shouldn't this and lines below be removed? they have a match with cName,tName above

File: matrix_70X.txt [hunk 4]
```diff
@@ -1,190 +1,190 @@
...
-RunMET2015Cwf134p805 134.805_RunMET2015C+RunMET2015C+HLTDR2_25ns+RECODR2_25nsreHLT+HARVESTDR2_25nsreHLT/step3.root reRECO
-RunDoubleEG2015Dwf134p902 134.902_RunDoubleEG2015D+RunDoubleEG2015D+HLTDR2_25ns+RECODR2_25nsreHLT+HARVESTDR2_25nsreHLT/step3.root reRECO
-RunDoubleMuon2015Dwf134p903 134.903_RunDoubleMuon2015D+RunDoubleMuon2015D+HLTDR2_25ns+RECODR2_25nsreHLT+HARVESTDR2_25nsreHLT/step3.root reRECO
-RunMET2015Dwf134p905 134.905_RunMET2015D+RunMET2015D+HLTDR2_25ns+RECODR2_25nsreHLT+HARVESTDR2_25nsreHLT/step3.root reRECO
-RunSinglePh2015Dwf134p911 134.911_RunSinglePh2015D+RunSinglePh2015D+HLTDR2_25ns+RECODR2_25nsreHLT+HARVESTDR2_25nsreHLT/step3.root reRECO
-TTbar14TeV2017wf10039p0 10039.0_TTbar_14TeV+TTbar_Tauola_14TeV_2017_GenSimFullINPUT+DigiFull_2017+RecoFull_2017/step3.root RECO
-MinBias14TeV2017wf10046p0 10046.0_MinBias_TuneZ2star_14TeV+MinBias_TuneZ2star_14TeV_pythia6_2017_GenSimFullINPUT+DigiFull_2017+RecoFull_2017/step3.root RECO
-TenMuE2017wf10060p0 10060.0_TenMuE_0_200+TenMuE_0_200_pythia8_2017_GenSimFullINPUT+DigiFull_2017+RecoFull_2017/step3.root RECO
+RunMinBias2011Awf1000p0 1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT step2.root RECO
```
Comment (slava77) [inline matrix_70X.txt]: I'm changing my mind on formatting of this file.
Similar to the DQM.sh file, for most cases the miniAOD file can be detected automatically: it is (almost) always a separate file with the same stepX and it is of the form stepX_inMINIAOD\*.root (SIM may or may not be there).
So, let's go back.

File: validateJR.sh [hunk 5]
```diff
@@ -3,34 +3,47 @@ baseA=$1
...
+        echo "Will run on ${dirN}/${mFN} in ${cWD}/${extmN}"
+        echo "Now in `pwd`"
+        echo -e "gSystem->Load(\"libFWCoreFWLite.so\");\n FWLiteEnabler::enable();\n 
+        .x validate.C+(\"${extmN}\", \"${baseA}/${dirN}/${mFN}\", \"${baseB}/${dirN}/${mFN}\", \"${mProcN}\");\n .qqqqqq" | root -l -b >& ${extmN}.log &
+    fi
+    #manually set the make flags, not needed in most cases
+    #gSystem->SetMakeSharedLib(\"cd \$BuildDir ;g++  -c \$Opt -pipe -m64 -Wshadow -Wall -W -Woverloaded-virtual -fPIC -std=c++11 -Wno-deprecated-declarations -DG__MAXSTRUCT=36000 -DG__MAXTYPEDEF=36000 -DG__LONGLINE=4096 -pthread \$IncludePath \$SourceFiles ; g++ \$ObjectFiles -shared -Wl,-soname,\$LibName.so -m64 -Wl,--hash-style=gnu -O2  \$LinkedLibs -o \$SharedLib\");\n 
+    #cout<< gSystem->GetMakeSharedLib()<<endl ;\n
 
     pidList=${pidList}" "${!}
```
Comment (slava77) [inline validateJR.sh]: this captures only the last started bg process.
the pidList and then sleep logic should be added after every bg process.
Can be copy-pasted, but may be better wrapped in a function.

**Comment by slava77** [comment]
@dntaylor what's your availability?
If you can not follow up on the comments in the next few days, I could take it over and address my comments.

**Pull request merged**